pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go

```diff
@@ -0,0 +1,213 @@
+	// clusters according to this order. When the system load decreases, the
+	// replicas will be scaled down from these clusters in reverse order.
+	//
+	// The candidate clusters will be preliminarily selected by the PropagationPolicy
+	// (or ClusterPropagationPolicy) and the scheduler's strategy, while ClusterScalePreferences
+	// further specifies the available cluster. If a cluster in the ClusterScalePreferences
+	// list but is excluded by the scheduler, it will be ignored.
+	// +required
+	ClusterScalePreferences []ClusterScalePreference `json:"clusterScalePreference"`
+}
+
+// CrossVersionObjectReference contains enough information to let you identify
+// the referred resource.
+type CrossVersionObjectReference struct {
+	// APIVersion represents the API version of the referent.
+	// +required
+	APIVersion string `json:"apiVersion"`
+
+	// Kind represents the Kind of the referent.
+	// +required
+	Kind string `json:"kind"`
+
+	// Name represents the name of the referent.
+	// +required
+	Name string `json:"name"`
+}
+
+// ClusterScalePreference describes customized behavior for specific cluster.
+type ClusterScalePreference struct {
+	// ClusterName is the name of the member cluster.
+	// +required
+	ClusterName string `json:"clusterName"`
+
+	// MinReplicas is the minimum number of replicas on the current cluster,
+	// Karmada will ensure that the number of replicas running on this cluster
+	// is not less than this value.
+	//
+	// Note that this only valid when at least one replica is running in the
+	// current cluster. See InitialReplicas for more details.
+	//
+	// It defaults to 1 pod.
+	// +optional
+	MinReplicas *int32 `json:"minReplicas,omitempty"`
+
+	// MaxReplicas is the maximum number of replicas allowed on the current
+	// cluster, Karmada will ensure that the number of running replicas does not
+	// exceed this value during coordination.
+	//
+	// Karmada suggests setting this value to ensure that replicas do not scale
+	// indefinitely, thus affecting other applications in the system.
+	//
+	// It cannot be less than MinReplicas.
+	// +required
+	MaxReplicas int32 `json:"maxReplicas"`
+
+	// InitialReplicas is the number of replicas to be placed in the current
+	// cluster during the initial stage.
+	//
+	// If the number is greater than zero, Karmada will place specified amount
+	// of replicas in the cluster and then delegate the replica management to
+	// the HPA running within the cluster.
+	//
+	// If the number is not set or set to 0, the cluster will act as a backup
+	// cluster and no replicas will be placed during the initial stage. Furthermore,
+	// the cluster might be activated when the load raised on other clusters, see
+	// Expansion for more details.
+	//
+	// The value must be greater than or equal to MinReplicas and less than or
+	// equal to MaxReplicas.
+	// +optional
+	InitialReplicas *int32 `json:"initialReplicas,omitempty"`
```

Comment:
Is `InitialReplicas` necessary? Could we use `MinReplicas` as the initial number of replicas instead?